src/environmentd/tests/sql.rs

```diff
@@ -1027,41 +1027,39 @@ fn test_temporary_views() -> Result<(), Box<dyn Error>> {
     Ok(())
 }
 
-// Test EXPLAIN TIMESTAMP with tables. Mock time to verify initial table since
-// is now(), not 0.
+// Test EXPLAIN TIMESTAMP with tables.
 #[test]
 fn test_explain_timestamp_table() -> Result<(), Box<dyn Error>> {
     mz_ore::test::init_logging();
-    let timestamp = Arc::new(Mutex::new(1_000));
-    let now = {
-        let timestamp = Arc::clone(&timestamp);
-        NowFn::from(move || *timestamp.lock().unwrap())
-    };
-    let config = util::Config::default().with_now(now);
+    let config = util::Config::default();
     let server = util::start_server(config)?;
     let mut client = server.connect(postgres::NoTls)?;
-    let timestamp_re = Regex::new(r"\s*(\d{4}|0) \(\d+-\d\d-\d\d \d\d:\d\d:\d\d.\d\d\d\)").unwrap();
+    let timestamp_re = Regex::new(r"\s*(\d+) \(\d+-\d\d-\d\d \d\d:\d\d:\d\d.\d\d\d\)").unwrap();
     let bool_re = Regex::new(r"true|false").unwrap();
 
     client.batch_execute("CREATE TABLE t1 (i1 INT)")?;
 
-    let expect = "          query timestamp:<TIMESTAMP>
-                    since:[<TIMESTAMP>]
-                    upper:[<TIMESTAMP>]
-         global timestamp:<TIMESTAMP>
-  can respond immediately: <BOOL>
+    let expect = "                query timestamp:<TIMESTAMP>
+          oracle read timestamp:<TIMESTAMP>
+largest not in advance of upper:<TIMESTAMP>
+                          upper:[<TIMESTAMP>]
+                          since:[<TIMESTAMP>]
+        can respond immediately: <BOOL>
+                       timeline: Some(EpochMilliseconds)
 
 source materialize.public.t1 (u1, storage):
-            read frontier:[<TIMESTAMP>]
-           write frontier:[<TIMESTAMP>]\n";
+                  read frontier:[<TIMESTAMP>]
+                 write frontier:[<TIMESTAMP>]\n";
 
     let row = client
         .query_one("EXPLAIN TIMESTAMP FOR SELECT * FROM t1;", &[])
         .unwrap();
     let explain: String = row.get(0);
     let explain = timestamp_re.replace_all(&explain, "<TIMESTAMP>");
+    // TODO: annoyingly "can respond immediately" does seem to change if you,
+    // say, inject a sleep. Is it supposed to be deterministic?qs
```

Comment:
For tables, it should always be true. If not then we may need to open an issue and look into that.

Reply:
opened https://github.com/MaterializeInc/materialize/issues/16115 to track